WcaOnRails/spec/requests/api_competitions_spec.rb

```diff
@@ -0,0 +1,360 @@
+                  id: "333-1",
+                  format: "a",
+                  timeLimit: nil,
+                  cutoff: nil,
+                  advancementCondition: nil,
+                },
+              ],
+            },
+            {
+              id: "222",
+              rounds: [
+                {
+                  id: "222-1",
+                  format: "a",
+                  timeLimit: nil,
+                  cutoff: nil,
+                  advancementCondition: nil,
+                },
+              ],
+            },
+            {
+              id: "333oh",
+              rounds: [
+                {
+                  id: "333oh-1",
+                  format: "a",
+                  timeLimit: nil,
+                  cutoff: nil,
+                  advancementCondition: nil,
+                },
+              ],
+            },
+          ]
+          patch api_v0_competition_update_events_from_wcif_path(competition), params: competition_events.to_json, headers: headers
+          expect(response).to have_http_status(422)
+          response_json = JSON.parse(response.body)
+          expect(response_json["error"]).to eq "Cannot add events to a confirmed competition"
+        end
+
+        it "does not allow removing events from a confirmed competition" do
+          headers = { "CONTENT_TYPE" => "application/json" }
+          competition_events = [
+            {
+              id: "333",
+              rounds: [
+                {
+                  id: "333-1",
+                  format: "a",
+                  timeLimit: {
+                    centiseconds: 4242,
+                    cumulativeRoundIds: [],
+                  },
+                  cutoff: nil,
+                  advancementCondition: nil,
+                },
+              ],
+            },
+            {
+              id: "222",
+              rounds: nil,
+            },
+          ]
+          patch api_v0_competition_update_events_from_wcif_path(competition), params: competition_events.to_json, headers: headers
+          expect(response).to have_http_status(422)
+          response_json = JSON.parse(response.body)
+          expect(response_json["error"]).to eq "Cannot remove events from a confirmed competition"
+        end
+      end
+
+      context "when signed in as a regular user" do
+        sign_in { FactoryBot.create :user }
+
+        it "does not allow access" do
+          patch api_v0_competition_update_events_from_wcif_path(competition)
+          expect(response).to have_http_status(403)
+          response_json = JSON.parse(response.body)
+          expect(response_json["error"]).to eq "Not authorized to manage competition"
+        end
+      end
+
+      describe "CSRF" do
+        # CSRF protection is always disabled for tests, enable it for this these requests.
+        before(:all) { ActionController::Base.allow_forgery_protection = true }
+        after(:all) { ActionController::Base.allow_forgery_protection = false }
```

Comment:
Nice! It might be nicer to use an `around` hook here instead. See https://relishapp.com/rspec/rspec-core/v/2-0/docs/hooks/around-hooks.

Reply:
Oh! I thought about doing a temporary configuration change with a block, but I ended up doing this. Great to see Rspec has this functionality on its own, thanks!